sdk/hasher/Cargo.toml

```diff
@@ -0,0 +1,32 @@
+[package]
+name = "solana-hasher"
+description = "Solana SHA256 hashing"
+documentation = "https://docs.rs/solana-hasher"
+version = { workspace = true }
+authors = { workspace = true }
+repository = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+solana-hash = { workspace = true }
+
+[target.'cfg(not(target_os = "solana"))'.dependencies]
+sha2 = { workspace = true }
+
+[target.'cfg(target_os = "solana")'.dependencies]
+sha2 = { workspace = true, optional = true }
```

Comment:
Just so I'm sure I understand, is the sha2 feature and this dependency specifically for the case where someone might want to try to hash on-chain without the syscall?

I'm not too sure of a use-case for that, so we might want to prefer removing the option totally. If someone really wants to do it, they can do it themselves by pulling in sha2 directly in their program. This seems like more of a footgun than anything else.

Or is this to avoid breaking people who might be depending on this currently? If that's the case, how about adding a comment saying to remove this line during the next breaking release?

Reply:
> Or is this to avoid breaking people who might be depending on this currently? If that's the case, how about adding a comment saying to remove this line during the next breaking release?

This, there's no use case I can conceive of but it's a breaking change